### PR TITLE
BUGFIX: Repair behat tests and Neos9-Beta-10 compatibility

### PR DESCRIPTION
**Pull request by mhsdesign**
The way how we write behat tests has undergone major changes via `neos/behat` and the runtime helpers in Neos as well.
For example we use workspace names now and dont block anymore.

Resolves: #76

### CODE REVIEW

**Comment by ahaeslich** [comment]
If I'm not misstaken we now need to adjust the tests to the workspace command change: https://github.com/neos/neos-development-collection/pull/4708

**Comment by mhsdesign** [comment]
Yes and that phpstan crashes the ci is also suboptimal

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -119,10 +119,11 @@ jobs:
         cd ${NEOS_BASE_FOLDER}
         ./flow cr:setup
 
-    - name: Run Functional tests
+    - name: Run Behavioral tests
       run: |
-        cd ${NEOS_BASE_FOLDER}
-        CATCHUPTRIGGER_ENABLE_SYNCHRONOUS_OPTION=1 bin/behat -c Packages/Application/Neos.RedirectHandler.NeosAdapter/Tests/Behavior/behat.yml.dist
+        cd ${NEOS_BASE_FOLDER}/bin
+        # FIXME we run in the bin dir directly as otherwise uribuilding is bugged: https://neos-project.slack.com/archives/C050KKBEB/p1718724415425439
```
Comment (mhsdesign) [inline .github/workflows/tests.yml:125]: will be fixed via https://github.com/neos/behat/pull/39

File: Classes/CatchUpHook/DocumentUriPathProjectionHook.php [hunk 2]
```diff
@@ -197,25 +195,21 @@ private function handleNodeWasMoved(NodeAggregateWasMoved $event, \Closure $clos
...
-
-                $closure($node, $this->getNodeAddress($event->contentStreamId, $newLocation->coveredDimensionSpacePoint, $node->getNodeAggregateId()), $this->contentRepository->id);
-
-                $descendantsOfNode = $this->getState()->getDescendantsOfNode($node);
-                array_map(fn (DocumentNodeInfo $descendantOfNode) => $closure(
-                    $descendantOfNode,
-                    $this->getNodeAddress($event->contentStreamId, $newLocation->coveredDimensionSpacePoint, $descendantOfNode->getNodeAggregateId()),
-                    $this->contentRepository->id
-                ), iterator_to_array($descendantsOfNode));
+        foreach ($event->succeedingSiblingsForCoverage as $interdimensionalSibling) {
```
Comment (mhsdesign) [inline Classes/CatchUpHook/DocumentUriPathProjectionHook.php:198]: Migrated in a similar fashion as the DocumentUriPathProjection here:
https://github.com/neos/neos-development-collection/pull/4993/files#diff-2e3df51a3340b51e08cf9d2ba5aade329a8c0e073354c1d5a72e42c3aaf5a127R618

**Pull request merged**